GlobalRegistrar/contract.sol

```diff
@@ -1,25 +1,65 @@
 //sol
 
-import "owned";
-
 contract NameRegister {
-	function addr(bytes32 _name) constant returns (address o_owner) {}
-	function name(address _owner) constant returns (bytes32 o_name) {}
+	function addr(bytes32 _name) constant returns (address o_owner);
+	function name(address _owner) constant returns (bytes32 o_name);
 }
 
 contract Registrar is NameRegister {
 	event Changed(bytes32 indexed name);
 	event PrimaryChanged(bytes32 indexed name, address indexed addr);
 
-	function owner(bytes32 _name) constant returns (address o_owner) {}
-	function addr(bytes32 _name) constant returns (address o_address) {}
-	function subRegistrar(bytes32 _name) constant returns (address o_subRegistrar) {}
-	function content(bytes32 _name) constant returns (bytes32 o_content) {}
+	function owner(bytes32 _name) constant returns (address o_owner);
+	function addr(bytes32 _name) constant returns (address o_address);
+	function subRegistrar(bytes32 _name) constant returns (address o_subRegistrar);
+	function content(bytes32 _name) constant returns (bytes32 o_content);
 	
-	function name(address _owner) constant returns (bytes32 o_name) {}
+	function name(address _owner) constant returns (bytes32 o_name);
+}
+
+contract AuctionSystem {
```

Comment:
I guess Registrar and NameRegister are meant to be plain interfaces (one to retrieve names and one to register them), but I actually do not know what the intention of the original author was. We can address this in a second PR.

Concerning the bidding function, I will try to clean it up.